src/libs/actions/Policy.ts

```diff
@@ -1611,12 +1612,14 @@ function buildOptimisticPolicyRecentlyUsedTags(policyID: string, tag: string): R
  *
  * @returns policyID of the workspace we have created
  */
-function createWorkspaceFromIOUPayment(iouReport: Report): string | undefined {
+function createWorkspaceFromIOUPayment(iouReportParam: Report | EmptyObject): string | undefined {
     // This flow only works for IOU reports
-    if (!ReportUtils.isIOUReport(iouReport)) {
+    if (!ReportUtils.isIOUReport(iouReportParam)) {
         return;
     }
 
+    const iouReport = iouReportParam as Report;
```

Comment:
> @ZhenjaHorbach @blazejkustra please you have time, could you give us advice here? I don't really like this way

Me neither haha

> Or is there any way for us to TS know that after call util ReportUtils.isIOUReport, it should be a Report type.

Yes, this should be possible! It's called [user defined type guard](https://levelup.gitconnected.com/user-defined-type-guards-in-typescript-fad639e4944f). We may need to create two different functions because of this (one for passing a report and one for passing reportID)

```ts
function isIOUReport(report: OnyxEntry<Report> | EmptyObject): report is Report {
    return report?.type === CONST.REPORT.TYPE.IOU;
}
```